addons/mail/models/mail_mail.py

```diff
@@ -123,6 +125,17 @@ def mark_outgoing(self):
     def cancel(self):
         return self.write({'state': 'cancel'})
 
+    @api.constrains('scheduled_date')
+    def _check_scheduled_date(self):
```

Comment:
After discussion, this should actually go back to a create / write override, as constraints are more used to raise if any error. Writing in a constraint is probably using a side effet of the ORM.